.github/actions/install_node_dependencies/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Setup Node.js and Install Dependencies
+description: Sets up a specific Node.js version, caches Node modules, and installs Node dependencies.
+
+inputs:
+  node-version:
+    description: 'Node.js version'
+    required: true
```

Comment:
I think we have a default value, so this should be optional right ?

Reply:
Yes, updated.